src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationOptions.cs

```diff
@@ -165,8 +165,14 @@ public AzureAppConfigurationOptions Select(string keyFilter, string labelFilter
             {
                 throw new ArgumentException("The characters '*' and ',' are not supported in label filters.", nameof(labelFilter));
             }
-
-            if (!_kvSelectors.Any(s => string.Equals(s.KeyFilter, keyFilter) && string.Equals(s.LabelFilter, labelFilter)))
+            var foundExactSameSelector = _kvSelectors.Find(s => string.Equals(s.KeyFilter, keyFilter) && string.Equals(s.LabelFilter, labelFilter));
+            if (foundExactSameSelector != null)
+            {
+                // Move to the end, keeping precedence.
+                _kvSelectors.Remove(foundExactSameSelector);
+                _kvSelectors.Add(foundExactSameSelector);
+            }
+            else
```

Comment:
Looks like we have the same problem while selecting feature flags or snapshots.

Reply:
Also, same problem when adding to `_multiKeyWatchers` list. None of them are preserving precedence order.

Reply:
Are you suggesting me to fix them together in this PR, or in a separate PR?
Or we extract a utility to perform deduplication of selectors/watchers?

Reply:
I think we can fix it in the same PR since the idea is the same. A common utility function might be good idea.

Reply:
OK

Reply:
MultiKeyWatchers has slightly different process while dedup, so I simply extracted a util for `List<KeyValueSelector>` instead of for both selectors and watchers. Let me know if it's ok...Otherwise I would have to add a generic utility, accepting detailed callbacks for existing/non-existing cases, which might make it a little bit complicated.